### ISSUE TITLE
Cycle consistency weight = 0

### ISSUE THREAD

**Issue by htzheng** [microsoft/CoCosNet#6]
In the provided training script, `opt.warp_cycle_w` is set to 0 by default. I wonder what is the weight for cycle correspondence regularization. Also, what is the good way to set flags for `opt.warp_patch` and `opt.two_cycle`?

Finally, is it possible to pretrain the correspondence module with the current code? Thanks!

**Comment by panzhang0212** [comment]
warp_cycle_w is 0.1 for edge-to-face, which help keep correspondence meaningful(figure 9 in our paper. In other dataset, you can set it to 0.1 to get better synthesis, but we find if this weight is too large, the correspondence will be identity mapping, causing  trivial solutions(exemplar image ≈ warped image). opt.warp_patch is True to keep better detail, for example, clothes pattern, so we use it for deepfashion dataset.
opt.two_cycle is not suggested to use. cycle1: A->B->A, cycle2:B->A->B. if your set it true, the reconstruction loss in cycle1 and cycle2 will be used. if your set it false only reconstruction loss in cycle2 will be used.
The correspondence module and translation module are joint training. So do not need to pretrain the correspondence module. I have tried pretraining the correspondence module, the performance is more poor than joint training.

**Comment by htzheng** [comment]
Thank you very much for the detailed explanation, I really appreciated it!

Also the flag `opt.noise_for_mask` is False by default, I wonder if  `opt.noise_for_mask` should be set to True for training ADE20k with warm-up strategy.

**Comment by panzhang0212** [comment]
in test phase, the inputs include input mask, exemplar natural image, exemplar mask. However, exemplar mask is not necessary, depending on training setting. If opt.noise_for_mask = false, training use three images as inputs, so test use three images as inputs.
If opt.noise_for_mask = true. training use three images as inputs during warmup, then use noise to replace exemplar mask and continue training, So test use two images(input mask, exemplar natural image) as inputs..

So opt.noise_for_mask is true or false, depending on whether you want use exemplar mask  as input in test phase.
